compiler/packages/babel-plugin-react-compiler/src/Validation/ValidateStaticComponents.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {CompilerError, ErrorSeverity} from '../CompilerError';
+import {HIRFunction, IdentifierId, SourceLocation} from '../HIR';
+import {Err, Ok, Result} from '../Utils/Result';
+
+/**
+ * Validates against components that are created dynamically and whose identity is not guaranteed
+ * to be stable (which would cause the component to reset on each re-render).
+ */
+export function validateStaticComponents(
+  fn: HIRFunction,
+): Result<void, CompilerError> {
```

Comment:
per feedback i switched to return a Result here. this lets us easily refactor later to throw the error if we want, without rewriting the validation pass itself